text/0249-v2-experiments.md

```diff
@@ -638,8 +638,71 @@ use a naming scheme that convey its non-final nature for the following reasons:
    that this is not the final version of the API, and its deprecation is
    expected.
 
+### Modules lifecycle
+
+This section discuss adding a new modules to `aws-cdk-lib`.
+
+#### Alpha (optional)
+
+This stage is intended for modules that are in early development stage, they
+change rapidly and may incurs many breaking changes. These modules will be
+released separately from `aws-cdk-lib` and will have their own version, which
+allows for breaking changes in accordance to semver, e.g `0.x` or prerelease
+qualifiers.
+
+In order to release a module as alpha, the following conditions must be met:
+
+1. Module from `aws-cdk-lib` **can not** depend on it.
```

Comment:
It's worth discussing the "integrations use case" (such as lambda-event-sources, sns-subscriptions). If those modules wish to include an integration with resource that is still in alpha, we will need to do something like only add the minimal bear interface to the stable module (i.e. `appmesh.IFoo` with only `fooArn` for example).

Worth mentioning this pattern because I think we will need to use it quite a lot in our transition from V1 to V2.